src/search.c

```diff
@@ -40,6 +41,121 @@ bool init = false;
 static char search_type = '0';
 static char scope = '0';
 
+struct bundle_result {
+	char bundle_name[256];
+	long size;
+	double score;
+	struct list *files;
+};
+
+struct file_result {
+	char filename[PATH_MAX+1];
+	double score;
+};
+
+static struct list *results;
+
+
+static void add_bundle_file_result(char *bundlename, char *filename, double score, struct manifest *man) 
+{
+	struct bundle_result *bundle = NULL;
+	struct file_result *file;
+	struct list *ptr;
+
+	ptr = results;
+	while (ptr) {
+		struct bundle_result *item;
+
+		item = ptr->data;
+		if (strcmp(item->bundle_name, bundlename) == 0) {
```

Comment:
Since we know the max size at all times is 256, why not go 1 step further and
`strncmp(item->bundle_name, bundlename, sizeof(item->bundle_name)) // so we don't hardcode in magic numbers`

Reply:
strncmp is usually slightly slower than strcmp, particularly if you allow strcmp to "overread".

Reply:
Is that really an issue in this code path...

Reply:
It is not an issue for this code path. I was just trying to understand why you should want to change strcmp(a,b) to strncmp(a,b,MAXSIZE).